tensorflow/core/kernels/data/fixed_length_record_dataset_op.cc

```diff
@@ -0,0 +1,512 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/core/kernels/data/fixed_length_record_dataset_op.h"
+
+#include "tensorflow/core/common_runtime/metrics.h"
+#include "tensorflow/core/framework/partial_tensor_shape.h"
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/kernels/data/name_utils.h"
+#include "tensorflow/core/lib/io/buffered_inputstream.h"
+#include "tensorflow/core/lib/io/inputbuffer.h"
+#include "tensorflow/core/lib/io/random_inputstream.h"
+#include "tensorflow/core/lib/io/zlib_compression_options.h"
+#include "tensorflow/core/lib/io/zlib_inputstream.h"
+
+namespace tensorflow {
+namespace data {
+
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kDatasetType;
+/* static */ constexpr const char* const FixedLengthRecordDatasetOp::kFileNames;
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kHeaderBytes;
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kRecordBytes;
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kFooterBytes;
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kBufferSize;
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kCompressionType;
+
+constexpr char kFixedLengthRecordDataset[] = "FixedLengthRecordDataset";
+constexpr char kCurrentFileIndex[] = "current_file_index";
+constexpr char kCurrentPos[] = "current_pos";
+constexpr char kZLIB[] = "ZLIB";
+constexpr char kGZIP[] = "GZIP";
+
+class FixedLengthRecordDatasetOp::Dataset : public DatasetBase {
+ public:
+  explicit Dataset(OpKernelContext* ctx, std::vector<string> filenames,
+                   int64 header_bytes, int64 record_bytes, int64 footer_bytes,
+                   int64 buffer_size, const string& compression_type,
+                   int op_version)
+      : DatasetBase(DatasetContext(ctx)),
+        filenames_(std::move(filenames)),
+        header_bytes_(header_bytes),
+        record_bytes_(record_bytes),
+        footer_bytes_(footer_bytes),
+        buffer_size_(buffer_size),
+        compression_type_(compression_type),
+        op_version_(op_version) {}
+
+  std::unique_ptr<IteratorBase> MakeIteratorInternal(
+      const string& prefix) const override {
+    name_utils::IteratorPrefixParams params;
+    params.op_version = op_version_;
+    if (compression_type_.empty()) {
+      return absl::make_unique<UncompressedIterator>(
+          UncompressedIterator::Params{
+              this, name_utils::IteratorPrefix(kDatasetType, prefix, params)});
```

Comment:
Note that the iterator prefix will be something like `PREFIX::FixedLengthRecordV2` for `FixedLengthRecordDatasetV2` Op.

Reply:
SGTM